Add Task to Poll for Ticket Status & to create new requests

zvercodebender opened the pull request:
Created a task to poll SN for field changes.  This could be useful in
blocking releases till they have been approved in SN.  Also added a
task to create new SN tickets.

Commit:
Merge remote-tracking branch 'xebialabs-community/master'

Commit:
Add new task types

Created a task to poll SN for field changes.  This could be useful in
blocking releases till they have been approved in SN.  Also added a
task to create new SN tickets.

jdewinne commented:
Would like to merge, but I would like to improve the code at the same time. Now all our tasks are having the same code to create connection etc... We should move that into a Jython ServiceNow class.

zvercodebender commented:
Can we merge this code, the work to consolidate the connection code into a ServiceNow class?  The consolidation of the code will not change the UI so this feature can be used while we work to improved the code.

Commit:
Refactor, remove included http class

Commit:
Merge branch 'sn-3.x' into xlr-5.x

Commit:
Merge branch 'xlr-5.x'

Commit:
typo in synthetic

Commit:
fix property format

Commit:
Testing in release flow

Commit:
some updates to the documentation

Commit:
move the bits around some

Commit:
add link to api docs

Commit:
Added a formatted output of records

Commit:
Added more error checking

Commit:
Updated more tasks.

zvercodebender commented:
This pull request refactores all of the current ServiceNow tasks as follows:
- Remove deprecated methods as follows:
  - CredentialsFallback
  - XLRequest
  
  These methods are replaced with new build in methods of the in the **ServiceNowClient** object.
- Removed the included **httputil** classes in favor of the new builtin [helper classes](https://docs.xebialabs.com/jython-docs/#!/xl-release/4.8.x/).
- Added more error checking
- Added output of ServiceNow requst objects in output log
- Added more documentation
- Included a XL Release tempalte to test current tasks
- Created default properties where possible
- Made **tableName** parameter hidden were possible
- Moved much of the detailed interactions into **ServiceNowClient** Class to integrate with ServiceNow

Commit:
Changed source of HttpRequest
